java/arcs/core/crdt/CrdtSingleton.kt

```diff
@@ -104,17 +103,40 @@ class CrdtSingleton<T : Referencable>(
     override fun hashCode(): Int = set.hashCode()
 
     /** Abstract representation of the data stored by a [CrdtSingleton]. */
-    interface Data<T : Referencable> : CrdtSet.Data<T> {
-        override fun copy(): Data<T>
+    interface Data<T : Referencable> : CrdtData {
+        val values: MutableMap<ReferenceId, CrdtSet.DataValue<T>>
+
+        /** Constructs a deep copy of this [Data]. */
+        fun copy(): Data<T>
+
+        /**
+         * Converts this instance into an equivalent instance of [CrdtSet.Data].
+         *
+         * CrdtSingleton is implemented via a backing CrdtSet, and their [Data] types are trivially
+         * convertible from one to the other.
+         */
+        fun asCrdtSetData(): CrdtSet.Data<T>
     }
 
     /** Concrete representation of the data stored by a [CrdtSingleton]. */
     data class DataImpl<T : Referencable>(
         override var versionMap: VersionMap = VersionMap(),
         override val values: MutableMap<ReferenceId, CrdtSet.DataValue<T>> = mutableMapOf()
     ) : Data<T> {
-        override fun copy(): Data<T> =
-            DataImpl(versionMap = VersionMap(versionMap), values = HashMap(values))
+        override fun asCrdtSetData() = CrdtSet.DataImpl(VersionMap(versionMap), HashMap(values))
```

Comment:
why not just
CrdtSet.DataImpl(versionMap, values) ?
(same question for RefModeCrdtTypes.kt)

Reply:
I had originally wanted to make a defensive copy in case someone tries to mutate it. But come to think of it now, if you call asCrdtSetData() and then mutate it, you probably want the changes to be reflected in the original...

Done.